code/modules/spells/spell_types/knock.dm

```diff
@@ -14,18 +14,5 @@
 
 /obj/effect/proc_holder/spell/aoe_turf/knock/cast(list/targets,mob/user = usr)
 	SEND_SOUND(user, sound('sound/magic/knock.ogg'))
-	for(var/turf/T in targets)
-		for(var/obj/machinery/door/door in T.contents)
-			INVOKE_ASYNC(src, .proc/open_door, door)
-		for(var/obj/structure/closet/C in T.contents)
-			INVOKE_ASYNC(src, .proc/open_closet, C)
-
-/obj/effect/proc_holder/spell/aoe_turf/knock/proc/open_door(obj/machinery/door/door)
-	if(istype(door, /obj/machinery/door/airlock))
-		var/obj/machinery/door/airlock/A = door
-		A.locked = FALSE
-	door.open()
-
-/obj/effect/proc_holder/spell/aoe_turf/knock/proc/open_closet(obj/structure/closet/C)
-	C.locked = FALSE
-	C.open()
+	for(var/turf/nearby_turf in targets)
+		SEND_SIGNAL(nearby_turf, COMSIG_ATOM_MAGICALLY_UNLOCKED, src, user)
```

Comment:
Should we really have the spell be an arg here? Without it, this could potentially be used elsewhere as some generic signal, as well (though it currently isnt a problem since it's unused).